test/cppsuite/tests/test_disagg_failover_perf.cpp

```diff
@@ -0,0 +1,593 @@
+          session.get(), ("commit_timestamp=" + timestamp_manager::decimal_to_hex(++ts)).c_str()));
+
+        /* Read the current value (optional, but as per prompt). */
+        cursor->set_key(cursor.get(), k.c_str());
+        /* All keys must exist. */
+        testutil_check(cursor->search(cursor.get()));
+        /* Overwrite with a new value. */
+        testutil_assert(crud::update(cursor, txn, k, generate_value()));
+        testutil_assert(txn.commit(session));
+        ingested_data += opt.key_size + opt.value_size;
+    }
+}
+
+static void
+crud_worker(workload_type type)
+{
+    scoped_session session = connection_manager::instance().create_session();
+    struct collection_cursor {
+        collection_cursor(collection &coll, scoped_cursor &&cursor)
+            : coll(coll), cursor(std::move(cursor))
+        {
+        }
+        scoped_cursor cursor;
+        collection &coll;
+    };
+    std::map<int, collection_cursor> cursor_map;
+    uint64_t ingested_data = 0;
+    uint64_t last_logged_mb = 0;
+    uint64_t target_bytes = opt.ingest_size_mb * 1000ULL * 1000ULL;
+    logger::log_msg(LOG_INFO,
+      "Ingest: Starting " + std::string(type == workload_type::append ? "append" : "update") +
+        " workload, target " + std::to_string(opt.ingest_size_mb) + "MB");
+    while (ingested_data < target_bytes) {
+        /* Generate a random int between 0 and collection count. */
+        int collection_num =
+          random_generator::instance().generate_integer(0, opt.collection_count - 1);
+        if (cursor_map.find(collection_num) == cursor_map.end()) {
+            collection &coll = database_model->get_collection(collection_num);
+            /*
+             * Construct the mapped value in-place. Using operator[] and then assigning
+             * requires the mapped_type to be assignable which is not true here because
+             * `collection_cursor` holds a reference member (and a move-only cursor). That
+             * deletes the implicit assignment operator. Emplace avoids assignment by
+             * constructing the value directly in the map.
+             */
+            cursor_map.emplace(
+              collection_num, collection_cursor(coll, session.open_scoped_cursor(coll.name)));
+        }
+        /*
+         * Access the stored scoped_cursor member from the mapped value. Use `at` to avoid
+         * accidental default-construction if the key were missing.
+         */
+        collection_cursor &cc = cursor_map.at(collection_num);
+        scoped_cursor &cursor = cc.cursor;
+        /* Workload logic. */
+        if (type == workload_type::append)
+            append(cc.coll, session, cursor, ingested_data);
+        else if (type == workload_type::update)
+            update(cc.coll, session, cursor, ingested_data);
+        /* Log every 100MB or 10% progress, whichever is less frequent. */
+        uint64_t current_mb = ingested_data / 1000 / 1000;
```

Comment:
can we use `( )` for clarity here?

Reply:
I'm not sure what you mean, this seems relatively clear to me.

Reply:
I guess this is more of a nit pick. Feel free to go with your preference. I meant like:
```
ingested_data / (1000 * 1000)
```
or even
```
bytes_per_mb = 1000 / 1000;
current_mb = ingested_data / bytes_per_mb;
```

Reply:
I will leave it as is in that case.